### DISCUSSION TITLE
Support for unordered snapsnot matching

### DISCUSSION THREAD

**Discussion by sazzer** [mitsuhiko/insta#294] [Q&A]
Hi there,

Is there any way to do snapshot matching where the order of object keys is not a concern? I know about `sort_keys`, but this just sorts the rendered JSON before comparing it to the snapshot which isn't quite the same thing.

In particular, I'm building a HAL implementation, and part of that I'm comparing the rendered output to the [example from the spec](https://datatracker.ietf.org/doc/html/draft-kelly-json-hal#section-6). Except that the example isn't strictly ordered. This is in ways that don't actually affect anything, but it does mean that I can't just paste the example as-is into the snapshot and have it work.

Probably what I want is something like `sort_keys` but working on the snapshot as well as the rendered JSON.

Cheers

**Comment by mitsuhiko** [comment]
Insta doesn’t understand the snapshot format. It does a string compare so all normalization needs to happen ahead of time.

I’m not sure what a good solution I can provide here. Maybe you could build yourself a script that sorts the JSON and then you paste that in.
